### ISSUE TITLE
Null-safety specification: Change LaTeX to markdown?

### ISSUE THREAD

**Issue by eernstg** [dart-lang/language#947]
With PR #941, several snippets of LaTeX (shown verbatim as source) were introduced, because the text specifies concrete edit operations on the language specification.

For improved readability, I believe we can change these snippets to markdown without introducing any significant amount of ambiguity.

If we wish to avoid the precise links into the language specification source code, we may then choose to extend the snippets such that they include more context (for instance, we could list all items in an itemized list rather than just the one that is being updated, such that it is unambiguous which item it is). This would make the null-safety spec more self-contained.

@leafpetersen, @lrhn, @munificent, WDYT?

**Comment by lrhn** [comment]
As for not using LaTeX source, yes please!

The specification is not specification *source*. Nobody reads the source. Changes to the specification should be described in terms of changes to the rendered output, not the source.

I'd probably just say something like;

> In section 3.9, The Flooing of Floos, the paragraph starting with "A Floo must 
> either be too early or too late" is changed to:
> > A *Floo* must neither have a time before *T*<sub>0</sub> nor one after *T*<sub>&infin;</sub>, even if it's a a <code>Future<*X*></code>.

It's possible to do quite elaborate markdown+html code if we want to get close to the LaTeX rendering.

**Comment by leafpetersen** [comment]
Yes.  In general I aim for the feature spec to be as self-contained as possible.  If necessary we can call out to the spec, but as noted this inevitably points to source, which is not what people should need to read.  This is one of the problems with a latex -> pdf spec: we can't link into the rendered version.

So in general, I prefer to either do as Lasse suggests, or to simply say "In the null safety release, The Flooing of the Floos works as follows:` followed by a complete specification of how to Floo a Floo.    This is definitely more work, but I results in the clearest and most readable result.

**Comment by munificent** [comment]
I mean, honestly, I'd be happy if the entire spec was written in Markdown. :)

**Comment by lrhn** [comment]
> I mean, honestly, I'd be happy if the entire spec was written in Markdown. :)

Hear, hear!
Not having a web-version of the spec is ... so last millenium!

**Comment by mraleph** [comment]
FWIW you can render LaTeX in the browser using for example MathJax and combine it with Markdown. 

Though enabling this in code review / GitHub interface would require writing a Chrome Extension.

**Comment by lrhn** [comment]
It's not the rendering which worries me, it's the lack of searchability. Even MathJax renders `$boo^2$` into something that I can't search for or select (at least if their demo is something to go by).

**Comment by eernstg** [comment]
I think we should have a way to access the language specification as a set of web pages, such that we can create links to specific locations that are readable (as opposed to the ones that we have used a lot of times, like [this one](https://github.com/dart-lang/language/blob/8cb7e46bfafca5356028b4c86ab1e1aad928a85d/specification/dartLangSpec.tex#L296)).

MathJax creates a bunch of code for each math element, and that's apparently completely unsearchable, and I would find that very, very inconvenient, too. Here is `$boo^2$` source:

```
<mjx-container class="MathJax CtxtMenu_Attached_0" jax="CHTML" role="presentation" tabindex="0" ctxtmenu_counter="38" style="font-size: 114.8%; position: relative;"><mjx-math class="MJX-TEX" aria-hidden="true"><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D44F TEX-I"></mjx-c></mjx-mi><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D45C TEX-I"></mjx-c></mjx-mi><mjx-msup><mjx-mi class="mjx-i"><mjx-c class="mjx-c1D45C TEX-I"></mjx-c></mjx-mi><mjx-script style="vertical-align: 0.363em;"><mjx-mn class="mjx-n" size="s"><mjx-c class="mjx-c32"></mjx-c></mjx-mn></mjx-script></mjx-msup></mjx-math><mjx-assistive-mml role="presentation" unselectable="on" display="inline"><math xmlns="http://www.w3.org/1998/Math/MathML"><mi>b</mi><mi>o</mi><msup><mi>o</mi><mn>2</mn></msup></math></mjx-assistive-mml></mjx-container>
```
